esmvalcore/preprocessor/_time.py

```diff
@@ -1292,3 +1298,424 @@ def resample_time(
         raise ValueError(
             f"Time coordinate {dates} does not contain {requested} for {cube}")
     return cube
+
+
+def _lin_pad(array: np.ndarray, delta: float, pad_with: int) -> np.ndarray:
+    """Linearly pad an array on both sides with constant difference."""
+    end_values = (array[0] - pad_with * delta, array[-1] + pad_with * delta)
+    new_array = np.pad(array, pad_with, 'linear_ramp', end_values=end_values)
+    return new_array
+
+
+def _guess_time_bounds(time_coord: DimCoord) -> None:
+    """Guess bounds of time coordinate in-place."""
+    if time_coord.has_bounds():
+        return
+    try:
+        time_coord.guess_bounds()
+    except ValueError:  # coordinate has only 1 point
+        point = time_coord.points[0]
+        time_coord.bounds = [[point - 0.5, point + 0.5]]
+
+
+def _get_lst_offset(lon_coord: Coord) -> np.ndarray:
+    """Get offsets to shift UTC time to local solar time (LST).
+
+    Note
+    ----
+    This function expects longitude in degrees. Can be in [0, 360] or [-180,
+    180] format.
+
+    """
+    # Make sure that longitude is in degrees and shift it to [-180, 180] first
+    # (do NOT overwrite input coordinate)
+    lon_coord = lon_coord.copy()
+    lon_coord.convert_units('degrees')
+    shifted_lon = (lon_coord.points + 180.0) % 360 - 180.0
+    return shifted_lon / 180.0 * 12.0
```

Comment:
```suggestion
    return (shifted_lon / 180.0) * 12.0
```
Just for readability of the calculation. Or put the 12 first for consistency with equation in the documentation.

Reply:
Changed inhttps://github.com/ESMValGroup/ESMValCore/pull/2258/commits/45024f7693e9f9afa2c8d77d47a87cb967f8c977